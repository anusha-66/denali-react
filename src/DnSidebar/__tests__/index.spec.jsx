/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnSidebar } from '../../index'; // src/index.tsx

describe('DnSidebar Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnSidebar>
                <DnSidebar.SectionHeader>SECTION HEADER</DnSidebar.SectionHeader>
                <DnSidebar.Item>Item 1</DnSidebar.Item>
                <DnSidebar.Item>Item 2</DnSidebar.Item>
                <DnSidebar.Item>Item 3</DnSidebar.Item>
            </DnSidebar>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnSidebar isTabletDownHideLeft className="dummy-sidebar">
                <DnSidebar.SectionHeader className="dummy-sidebar-section-header">
                    SECTION HEADER
                </DnSidebar.SectionHeader>
                <DnSidebar.Item className="dummy-sidebar-item">Item 1</DnSidebar.Item>
                <DnSidebar.Item isActive>Item 2</DnSidebar.Item>
                <DnSidebar.Item isDisabled>Item 3</DnSidebar.Item>
                <DnSidebar.Item>Item 4</DnSidebar.Item>
            </DnSidebar>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
