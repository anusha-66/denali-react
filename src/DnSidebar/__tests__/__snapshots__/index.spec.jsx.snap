// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnSidebar Tests should render correctly with custom props 1`] = `
<div
  class="tabs is-primary is-vertical tablet-down-hide-left dummy-sidebar"
>
  <ul>
    <li
      class="tabs__section-header dummy-sidebar-section-header"
    >
      <span>
        SECTION HEADER
      </span>
    </li>
    <li
      class="dummy-sidebar-item"
    >
      <span>
        Item 1
      </span>
    </li>
    <li
      class="is-active"
    >
      <span>
        Item 2
      </span>
    </li>
    <li
      class="is-disabled"
    >
      <span>
        Item 3
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Item 4
      </span>
    </li>
  </ul>
</div>
`;

exports[`DnSidebar Tests should render with default props 1`] = `
<div
  class="tabs is-primary is-vertical"
>
  <ul>
    <li
      class="tabs__section-header"
    >
      <span>
        SECTION HEADER
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Item 1
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Item 2
      </span>
    </li>
    <li
      class=""
    >
      <span>
        Item 3
      </span>
    </li>
  </ul>
</div>
`;
